wled00/data/settings_wifi.htm

```diff
@@ -70,37 +71,70 @@
 						option.setAttribute("value", networks[i].ssid);
 						option.textContent = `${networks[i].ssid} (${networks[i].rssi} dBm)`;
 
-						if (networks[i].ssid === cs.value) {
+						if (networks[i].ssid === input.value) {
 							option.setAttribute("selected", "selected");
+							found = true;
 						}
 
 						select.appendChild(option);
 					}
 					const option = cE("option");
 
 					option.setAttribute("value", "!Cs");
-					option.textContent = `Other network...`;
+					option.textContent = "Other network...";
 					select.appendChild(option);
 
-					cs.replaceWith(select);
+					if (input.value === "" || found) input.replaceWith(select);
+					else select.remove(); 
 				}
 
 				button.disabled = false;
 				button.textContent = "Scan";
 			});
 		}
 		// replace WiFi select with custom SSID input field again
-		function T() {
-			let cs = gId("CS");
+		function T(cs) {
 			if (!cs || cs.value != "!Cs") return;
 			let input = cE("input");
 			input.type = "text";
-			input.id = "CS";
-			input.name ="CS";
+			input.id = cs.id;
+			input.name = cs.name;
 			input.setAttribute("maxlength",32);
 			input.value = preScanSSID;
 			cs.replaceWith(input);
 		}
+		function resetWiFi(maxN = undefined) {
+			if (maxN) maxNetworks = maxN;
+			let entries = gId("wifi_entries").children
+			for (let i = entries.length; i > 0; i--) entries[i-1].remove();
+			btnWiFi(0);
+		}
+		function btnWiFi(i) {
+			gId("wifi_add").style.display = (i<maxNetworks) ? "inline":"none";
+			gId("wifi_rem").style.display = (i>1) ? "inline":"none";
+		}
+		function addWiFi(ssid="",pass="",ip=0,gw=0,sn=0x00ffffff) { // little endian
+			var i = gId("wifi_entries").childNodes.length;
+			if (i >= maxNetworks) return;
+			var b = `<div id="net${i}"><hr class="sml">
+Network name (SSID${i==0?", empty to not connect":""}):<br><input type="text" id="CS${i}" name="CS${i}" maxlength="32" value="${ssid}" ${i>1?"required":""}><br>
```

Comment:
The part of `${i>1?"required":""}` should be `${i>=1?"required":""}` because CS0 is the main WiFi, while CS1 and up are the "optional" ones.
Without that, the second WiFi's SSID can be set to empty